graphql_compiler/post_processing/sql_post_processing.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import html
+import re
+from typing import Any, Dict, List, Optional, Sequence
+
+from graphql import GraphQLList, GraphQLScalarType
+
+from ..compiler.compiler_frontend import OutputMetadata
+
+
+def _mssql_xml_path_string_to_list(
+    xml_path_result: str, list_entry_type: GraphQLScalarType
+) -> List[Any]:
+    """Convert the string result produced with XML PATH for MSSQL folds to a list.
+
+    Args:
+        xml_path_result: str, result from an XML PATH folded output
+        list_entry_type: GraphQLScalarType, type the results should be output as
+
+    Returns:
+        list representation of the result with all XML and GraphQL Compiler escaping reversed
+    """
+    # Return an empty list if the XML PATH result is "".
+    if xml_path_result == "":
+        return []
+
+    # Split the XML path result on "|".
+    list_result: Sequence[Optional[str]] = xml_path_result.split("|")
+
+    # Convert "~" to None. Note that this must be done before "^n" -> "~".
+    list_result = [None if result == "~" else result for result in list_result]
+
+    # Convert "^d" to "|".
+    list_result = [
+        result.replace("^d", "|") if result is not None else None for result in list_result
+    ]
+
+    # Convert "^n" to "~".
+    list_result = [
+        result.replace("^n", "~") if result is not None else None for result in list_result
+    ]
+
+    # Convert "^e" to "^". Note that this must be done after the caret escaped characters i.e.
+    # after "^n" -> "~" and "^d" -> "|".
+    list_result = [
+        result.replace("^e", "^") if result is not None else None for result in list_result
+    ]
+
+    # Convert "&#x{2 digit HEX};" to unicode character.
+    new_list_result: List[Optional[str]] = []
+    for result in list_result:
+        if result is not None:
+            split_result = re.split("&#x([A-Fa-f0-9][A-Fa-f0-9]);", result)
+            new_result = split_result[0]
+            for hex_value, next_substring in zip(split_result[1::2], split_result[2::2]):
+                new_result += chr(int(hex_value, 16)) + next_substring
+            new_list_result.append(new_result)
+        else:
+            new_list_result.append(None)
+
+    # Convert "&amp;" to "&", "&gt;" to ">", "&lt;" to "<". Note that the ampersand conversion
+    # must be done after the ampersand escaped HEX values.
+    list_result = [
+        html.unescape(result) if result is not None else None for result in new_list_result
+    ]
+
+    # Convert to the appropriate return type.
+    list_result_to_return: List[Optional[Any]] = [
+        list_entry_type.parse_value(result) if result is not None else None
+        for result in list_result
+    ]
+
+    return list_result_to_return
+
+
+def post_process_mssql_folds(
+    query_results: List[Dict[str, Any]], output_metadata: Dict[str, OutputMetadata]
+) -> None:
+    r"""Convert XML PATH fold results from a string to a list of the appropriate type.
```

Comment:
Sneaky `r`

Reply:
Needs to be a raw string (the `\xHEX` gives a `SyntaxError` otherwise) 😄

Reply:
That's kinda hilarious. Not sure if warrants a comment but it definitely sticks out a little.